gpflow/likelihoods/scalar_continuous.py

```diff
@@ -61,7 +61,11 @@ def _conditional_variance(self, F):
         return tf.fill(tf.shape(F), tf.squeeze(self.variance))
 
     def _predict_mean_and_var(self, Fmu, Fvar):
-        return tf.identity(Fmu), Fvar + self.variance
+        rank = tf.rank(Fvar).numpy()
+        if rank == 2:
+            return tf.identity(Fmu), Fvar + self.variance
+        else:
```

Comment:
Can we, please, have an `elif` that matches the exact ranks you want to target here, and another `else` that raises a `NotImplementedError` for the ones we do not support?